### PR TITLE
#659: Add completion for single quotation mark

### PR DESCRIPTION
**Pull request by telezhnaya**
I do not understand how to make tests for this changes. It is even not a simple completion - it's auto completion.
So, if tests are not necessary - great.
Else - help me please.

### CODE REVIEW

**Comment by deadok22** [comment]
com.intellij.testFramework.fixtures.CodeInsightTestFixture#type(char)

**Comment by telezhnaya** [comment]
Thank you.
Check it please.

**Comment by deadok22** [comment]
Please, try to make it work for the following scenario:
```erlang
'AtomNameWithCapitalLetters<caret>
```

**Comment by telezhnaya** [comment]
Check it please.

**Comment by deadok22** [comment]
```
java.lang.IndexOutOfBoundsException: Wrong index: -1
	at com.intellij.openapi.editor.ex.util.SegmentArrayWithData.getSegmentData(SegmentArrayWithData.java:103)
	at com.intellij.openapi.editor.ex.util.LexerEditorHighlighter$HighlighterIteratorImpl.getTokenType(LexerEditorHighlighter.java:391)
	at org.intellij.erlang.ErlangQuoteHandler.isOpeningQuote(ErlangQuoteHandler.java:32)
	at com.intellij.codeInsight.editorActions.TypedHandler.isOpeningQuote(TypedHandler.java:518)
	at com.intellij.codeInsight.editorActions.TypedHandler.handleQuote(TypedHandler.java:480)
	at com.intellij.codeInsight.editorActions.TypedHandler.access$300(TypedHandler.java:64)
	at com.intellij.codeInsight.editorActions.TypedHandler$1.perform(TypedHandler.java:204)
	at com.intellij.openapi.editor.impl.CaretModelImpl$3.run(CaretModelImpl.java:342)
	at com.intellij.openapi.editor.impl.CaretModelImpl.doWithCaretMerging(CaretModelImpl.java:422)
	at com.intellij.openapi.editor.impl.CaretModelImpl.runForEachCaret(CaretModelImpl.java:333)
	at com.intellij.openapi.editor.impl.CaretModelImpl.runForEachCaret(CaretModelImpl.java:324)
	at com.intellij.codeInsight.editorActions.TypedHandler.execute(TypedHandler.java:152)
	at com.intellij.codeInsight.lookup.impl.LookupTypedHandler.execute(LookupTypedHandler.java:91)
	at com.intellij.openapi.editor.actionSystem.TypedAction$TypingCommand$1.run(TypedAction.java:129)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:931)
	at com.intellij.openapi.editor.actionSystem.TypedAction$TypingCommand.run(TypedAction.java:123)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:124)
	at com.intellij.openapi.editor.actionSystem.TypedAction.actionPerformed(TypedAction.java:107)
	at com.intellij.openapi.editor.impl.EditorImpl.processKeyTyped(EditorImpl.java:1085)
	at com.intellij.openapi.editor.impl.EditorImpl.processKeyTyped(EditorImpl.java:5215)
	at com.intellij.openapi.editor.impl.EditorImpl$10.keyTyped(EditorImpl.java:915)
	at java.awt.Component.processKeyEvent(Component.java:6480)
	at javax.swing.JComponent.processKeyEvent(JComponent.java:2832)
	at java.awt.Component.processEvent(Component.java:6302)
	at java.awt.Container.processEvent(Container.java:2234)
	at java.awt.Component.dispatchEventImpl(Component.java:4881)
	at java.awt.Container.dispatchEventImpl(Container.java:2292)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1954)
	at java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:806)
	at java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1074)
	at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:945)
	at java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:771)
	at java.awt.Component.dispatchEventImpl(Component.java:4752)
	at java.awt.Container.dispatchEventImpl(Container.java:2292)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:734)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:539)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

```

**Comment by telezhnaya** [comment]
Check it please.

File: src/org/intellij/erlang/ErlangQuoteHandler.java [hunk 1]
```diff
@@ -17,17 +17,26 @@
...
 
   @Override
-  public boolean hasNonClosedLiteral(Editor editor, HighlighterIterator iterator, int offset) {
-    return true;
+  public boolean isOpeningQuote(HighlighterIterator iterator, int offset) {
+    if(iterator.getTokenType() == ErlangTypes.ERL_SINGLE_QUOTE) {
+      iterator.retreat();
+      boolean isAtomBefore = !iterator.atEnd() && iterator.getTokenType().equals(ErlangTypes.ERL_ATOM_NAME);
+      iterator.retreat();
+      boolean atomStartsFromQuote = !iterator.atEnd() && iterator.getTokenType().equals(ErlangTypes.ERL_SINGLE_QUOTE);
```
Comment (deadok22) [inline src/org/intellij/erlang/ErlangQuoteHandler.java]: Use equality comparison for token types.

**Comment by telezhnaya** [comment]
Check it please.

**Pull request merged**